common/app-starters-micrometer-common/src/test/java/org/springframework/cloud/stream/app/micrometer/common/AbstractMicrometerTagTest.java

```diff
@@ -58,6 +67,85 @@ public void before() {
 		assertNotNull("The jvm.memory.committed meter mast be present in SpringBoot apps!", meter);
 	}
 
+	@BeforeClass
+	public static void setup() {
+		CfEnvTestUtils.mockVcapServicesFromString(getServicesPayload(getMockVcap("cloud-test-info.json")));
+	}
+
+	private static String getMockVcap(String filename) {
+		return readTestDataFile(filename);
+	}
+
+	private static String readTestDataFile(String fileName) {
+		Scanner scanner = null;
+		try {
+			Reader fileReader = new InputStreamReader(
+					AbstractMicrometerTagTest.class.getResourceAsStream(fileName));
+			scanner = new Scanner(fileReader);
+			return scanner.useDelimiter("\\Z").next();
+		}
+		finally {
+			if (scanner != null) {
+				scanner.close();
+			}
+		}
+	}
+
+	private static String getServicesPayload(String... servicePayloads) {
+		Map<String, List<String>> labelPayloadMap = new HashMap<>();
+
+		for (String payload : servicePayloads) {
+			String label = getServiceLabel(payload);
+
+			List<String> payloadsForLabel = labelPayloadMap.computeIfAbsent(label, k -> new ArrayList<>());
+			payloadsForLabel.add(payload);
+		}
+
+		StringBuilder result = new StringBuilder("{\n");
+		int labelSize = labelPayloadMap.size();
+		int i = 0;
+
+		for (Map.Entry<String, List<String>> entry : labelPayloadMap.entrySet()) {
+			result.append(quote(entry.getKey())).append(":");
+			result.append(getServicePayload(entry.getValue()));
+			if (i++ != labelSize - 1) {
+				result.append(",\n");
+			}
+		}
+		result.append("}");
+		return result.toString();
+	}
+
+	private static String getServicePayload(List<String> servicePayloads) {
+		StringBuilder payload = new StringBuilder("[");
+
+		for (int i = 0; i < servicePayloads.size(); i++) {
+			payload.append(servicePayloads.get(i));
+			if (i != servicePayloads.size() - 1) {
+				payload.append(",");
+			}
+		}
+		payload.append("]");
+
+		return payload.toString();
+	}
+
+	private static String quote(String str) {
+		return "\"" + str + "\"";
+	}
+
+	@SuppressWarnings("unchecked")
+	private static String getServiceLabel(String servicePayload) {
+		try {
+			Map<String, Object> serviceMap = objectMapper.readValue(servicePayload,
+					Map.class);
+			return serviceMap.get("label").toString();
+		}
+		catch (Exception e) {
+			return null;
+		}
+	}
+
```

Comment:
We don't need this  json re-writing code. Instead we can change our test json to something like this:
```
{
  "sso":[{
    "name": "sso",
    "label": "sso",
    "plan": "notfree",
    "tags": ["configuration"],
    "credentials":{
      "uri": "http://pivotal.io",
      "client_id": "fakeClientId",
      "client_secret": "fakeSecret",
      "access_token_uri": "token"
    }
  }]}
``` 
And then add it to the env variables like this:
```
	@BeforeClass
	public static void setup() throws IOException {
		String serviceJson = StreamUtils.copyToString(new DefaultResourceLoader().getResource(
				"classpath:/org/springframework/cloud/stream/app/micrometer/common/pcf-scs-info.json")
				.getInputStream(), Charset.forName("UTF-8"));
		CfEnvTestUtils.mockVcapServicesFromString(serviceJson);
	}
```
No extra code is required